### PR TITLE
Add option to read pkeyopts interactively

### PR DESCRIPTION
**Pull request by johndoe31415**
This patch adds the ability to interactively enter passphrases for
the pkeyutl application. As requested by @mattcaswell in https://github.com/openssl/openssl/pull/3987, it is a cherrypicked commit that was originally included there.

Illustration of functionality: For example, you could use

$ openssl pkeyutl -kdf TLS1-PRF -kdflen 8 -pkeyopt md:md5
  -pkeyopt_passin secret -pkeyopt_passin seed

To have the "secret" and "seed" values read interactively from keyboard
(with hidden input). Alternatively, the pass phrase argument syntax is
also supported, e.g.:

$ openssl pkeyutl -kdf TLS1-PRF -kdflen 8 -pkeyopt md:md5
  -pkeyopt_passin secret:stdin -pkeyopt_passin seed:env:SEEDVAR

To have "secret" read from stdin and "seed" from the environment
variable SEEDVAR.

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: apps/pkeyutl.c [hunk 1]
```diff
@@ -232,6 +243,45 @@ int pkeyutl_main(int argc, char **argv)
...
+
+        for (i = 0; i < num; i++) {
+            char *opt = sk_OPENSSL_STRING_value(pkeyopts_passin, i);
+            char *passin = strchr(opt, ':');
+            char *passwd;
+
+            if (passin == NULL) {
+                /* Get password interactively */
+                char passwd_buf[4096];
+                snprintf(passwd_buf, sizeof(passwd_buf), "Enter %s: ", opt);
```
Comment (mattcaswell) [inline apps/pkeyutl.c]: Use BIO_snprintf

**Comment by mattcaswell** [comment]
We won't be able to push this at the moment. The master branch is currently frozen against new features.

**Comment by richsalz** [comment]
"at the moment" is a little bit of understatement. :) It won't show up until the next release after 1.1.1

**Comment by paulidale** [comment]
What is holding this one back?

**Comment by levitte** [comment]
Nothing tangible, other than lost in the noise.

**Comment by levitte** [comment]
Merged.

6dfcea3dfb167ccf37907601a3be6dc50f9c213b Add documentation for -pkeyopt_passin
9d1bf5f7dee88e2fc1149be16256404265765894 Add option to read pkeyopts interactively

**Comment by levitte** [comment]
(light editorial work done)

**Pull request closed without merging**